includes/plugin-instrumentation.php

```diff
@@ -30,11 +30,43 @@ function razorpayPluginActivated()
 
         $this->rzpTrackDataLake('plugin activate', $activateProperties);
 
+        $this->hposInstrumentation();
+
         $this->initRzpCronJobs();
 
         return 'success';
     }
 
+    function hposInstrumentation()
+    {
+        if (OrderUtil::custom_orders_table_usage_is_enabled() and
+            (empty(get_option('rzp_hpos')) or
+            get_option('rzp_hpos') === 'no'))
+        {
+            $properties = [
+                'isHposEnabled' => true
+            ];
+
+            $response = $this->rzpTrackSegment('hpos.interacted', $properties);
+            $this->rzpTrackDataLake('hpos.interacted', $properties);
+
+            update_option('rzp_hpos', 'yes');
+        }
+        else if(OrderUtil::custom_orders_table_usage_is_enabled() === false and
+                empty(get_option('rzp_hpos')) === false and
```

Comment:
get_option('rzp_hpos') === 'yes' condition will do job of empty(get_option('rzp_hpos')) === false. a condition can be removed